ResearchKit/ActiveTasks/ORKRecorder.m

```diff
@@ -274,7 +274,7 @@ @implementation ORKDataLogRecorder {
 @synthesize logger = _logger;
 
 - (ORKDataLogger *)logger {
-    return self.sharedLogger ? : _logger;
+    return [self isConsolidated] && (self.sharedLogger != nil) ? self.sharedLogger : _logger;
```

Comment:
yup.  It is currently redundant, *but* it is a two-part check. In other words, if the given recorder supports the shared logger (not all do) *and* the shared logger is non-nil, then return the shared logger. Otherwise, return `_logger`.

It's pretty obfuscated (sorry), but it was the best way I could figure to consolidate accelerometer, device motion and location into a single file that is using a shared timestamp.